_script/cibuild-docker

```diff
@@ -43,40 +43,44 @@ function arguments_mgmt() {
     shift "$((OPTIND-1))"
 }
 
-function travis_deploy_flags() {
-    docker_deploy_flags=""
-    if $(test "$sim_deploy" = "yes")
+function activate_deploy_flags() {
+    echo "INFO: deploy flag enabled. We will activate the env variables, and mount your docker config/credentials,"\
+         "to simulate a travis merge to master"
+    local_docker_config_store="${HOME}/.docker"
+    if ! $(test -f "${local_docker_config_store}/config.json")
     then
-        echo "INFO: deploy flag enabled. We will activate the env variables, and mount your docker config/credentials,"\
-             "to simulate a travis merge to master"
-        local_docker_config_store="${HOME}/.docker"
-        if ! $(test -f "${local_docker_config_store}/config.json")
-        then
-            echo "ERROR: You need a valid docker config credentials file in your ${local_docker_config_store} directory"\
-                 "in order to deploy new images, with read access to the user ${travis_user} (uid 2000 in the container)."\
-                 "You can generate it using docker login"
-            exit 1
-        fi
-        docker_deploy_flags="-v ${local_docker_config_store}:${travis_home}/.docker -e TRAVIS_PULL_REQUEST=false -e TRAVIS_BRANCH=master"
+        echo "ERROR: You need a valid docker config credentials file in your ${local_docker_config_store} directory"\
+             "in order to deploy new images, with read access to the user ${travis_user} (uid 2000 in the container)."\
+             "You can generate it using docker login"
+        exit 1
     fi
+    deploy_flags="-v ${local_docker_config_store}:${travis_home}/.docker -e TRAVIS_PULL_REQUEST=false -e TRAVIS_BRANCH=master"
 }
 
 function docker_run_cmdline_generator() {
     script_name=$(echo $(basename $0) | cut -d "-" -f 1)
     # https://docs.travis-ci.com/user/common-build-problems/#Troubleshooting-Locally-in-a-Docker-Image
     test_docker_image="travisci/ci-garnet:packer-1503972846"
-    travis_home="/home/travis"
     travis_user="travis"
-    travis_deploy_flags
+    travis_home="/home/${travis_user}"
+    if $(test "$sim_deploy" = "yes")
+    then
+        activate_deploy_flags
+    else
+        deploy_flags=""
+    fi
     docker_mounted_project="${travis_home}/gopath/src/github.com/dachad/tcpgoon"
     http_proxy="${http_proxy:-}"
-    cmdline="docker run -t -i -v $PWD:$docker_mounted_project \
+    cmdline="docker run -t -i \
+            -v $PWD:$docker_mounted_project \
```

Comment:
what's the purpose of mounting this volume? retrieving the coverall reports?

Reply:
well, in the docker run, this is the directory with the project / source code, so the docker image sees the same you have in the copy of the repo